src/redempt/redlib/region/SpheroidRegion.java

```diff
@@ -383,7 +384,34 @@ public boolean surfaceContains(Block block) {
 		}
 		return surface.contains(block);
 	}
-	
+
+	/**
+	 * Gets a random block within a spherical region
+	 * @return Bukkit block object of the randomly picked block
+	 */
+	public Block getRandomBlock() {
+		int[] randomBlockLocation = getRandomBlockCoordinates();
+		Location pickedLocation = new Location(getWorld(), randomBlockLocation[0], randomBlockLocation[1], randomBlockLocation[2]);
+		while (!contains(pickedLocation)) {
+			pickedLocation = new Location(getWorld(), randomBlockLocation[0], randomBlockLocation[1], randomBlockLocation[2]);
+		}
+		return pickedLocation.getBlock();
+	}
+
+	/**
+	 * Picks a random block coordinate from a sphere region,
+	 * this treats the sphere as cuboid, be sure to discard points outside the sphere
+	 * @return Array of three integers, one for each coordinate axis
+	 */
+	private int[] getRandomBlockCoordinates(){
+		Random random = new Random();
```

Comment:
Same here, make it take the Random instance as a parameter and add a no-param variant that uses ThreadLocalRandom.